### PR TITLE
When explaining execution reasons, add 'more...' if any reasons are omitted

### PR DESCRIPTION
**Pull request by mathjeff**
### Context

This might help a little bit with https://issuetracker.google.com/issues/321949384
It was originally surprising that Gradle wasn't listing all of the files in the directory

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team.
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by ov7a** [comment]
Thank you for your contribution!

Please add tests to cover your changes. 

***

The relevant team for this area will confirm the design of the implementation choices.

File: platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/changes/MessageCollectingChangeVisitorsTest.groovy [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.history.changes
+
+import spock.lang.Specification
+
+class MessageCollectingChangeVisitorsTest extends Specification {
```
Comment (mathjeff) [inline platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/changes/MessageCollectingChangeVisitorsTest.groovy:21]: I noticed that if I named this class `MessageCollectingChangeVisitorTest` in `MessageCollectingChangeVisitorTest.groovy` (what I wanted to call it) then it didn't report any failures and probably didn't run, so I added an `s` into the name of the test to get it to run. Does this seem like something we should be concerned about?

**Comment by mathjeff** [comment]
> Thank you for your contribution!
> 
> Please add tests to cover your changes.
> 
> The relevant team for this area will confirm the design of the implementation choices.

Thanks! I added some tests into the commit

File: platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/changes/MessageCollectingChangeVisitorsTest.groovy (hunk 1 above)
Comment (lptr) [reply platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/changes/MessageCollectingChangeVisitorsTest.groovy:21]: The name of the class should match the file name, but even if that's not the case, the tests should run. I tried renaming a similar test class and the tests in it kept executing fine, as they should.

Comment (mathjeff) [reply platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/changes/MessageCollectingChangeVisitorsTest.groovy]: I think I found a repro for the problem I was encountering about the test not running: https://github.com/gradle/gradle/issues/28916

In the meanwhile it appears that if I delete my Gradle cache that the test starts running again, so I renamed the test and its test file back to what I originally wanted to call it

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Comment by asodja** [comment]
@bot-gradle test QF

**Comment by asodja** [comment]
@bot-gradle test QF

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Review by lptr — APPROVED** [review]
LGTM.

**Comment by mathjeff** [comment]
Thanks!

**Pull request merged**